tests/integration_tests/scheduler/test_generic_driver.py

```diff
@@ -212,3 +213,33 @@ async def test_num_cpu_sets_env_variables(driver: Driver, tmp_path, job_name):
     elif isinstance(driver, OpenPBSDriver):
         assert "OMP_NUM_THREADS=2" in env_lines
         assert "NCPUS=2" in env_lines
+
+
+@pytest.mark.integration_test
+async def test_execute_with_retry_exits_on_filenotfounderror(driver: Driver, caplog):
+    caplog.set_level(logging.DEBUG)
+    invalid_cmd = ["/usr/bin/foo", "bar"]
+    (succeeded, message) = await driver._execute_with_retry(
+        invalid_cmd, total_attempts=3
+    )
+
+    assert "retry" not in str(caplog.text)
+    assert not succeeded
+    assert "No such file or directory" in message
+
+
+@pytest.mark.integration_test
+async def test_poll_exits_on_filenotfounderror(driver: Driver, caplog):
+    if isinstance(driver, LocalDriver):
+        pytest.skip("LocalDriver does not poll")
+    caplog.set_level(logging.DEBUG)
+    invalid_cmd = ["/usr/bin/foo", "bar"]
+    driver._bjobs_cmd = invalid_cmd
+    driver._qstat_cmd = invalid_cmd
+    driver._squeue = invalid_cmd
+    driver._jobs = {"foo": "bar"}
+    driver._non_finished_job_ids = ["foo"]
+    await driver.poll()
+
+    assert "retry" not in str(caplog.text)
```

Comment:
Don't understand this assert, maybe a comment that explains it (the one above too).